test/test_type_promotion.py

```diff
@@ -1,40 +1,56 @@
+from functools import wraps
+
 import torch
-import unittest
 import itertools
 
 from common_utils import TestCase, run_tests, load_tests
+from common_device_type import instantiate_device_type_tests
 
 # load_tests from common_utils is used to automatically filter tests for
 # sharding on sandcastle. This line silences flake warnings
 load_tests = load_tests
 
-class TestTypePromotion(TestCase):
+# Not thread-safe decorator that runs the decorated test once with
+# the default dtype being torch.float and again with the default dtype
+# being torch.double.
+def multiple_default_dtypes(fn):
```

Comment:
I thought we were going to call this float_double_default_dtype?

Reply:
Right; thanks for reminding me.